### PR TITLE
feat!: console usage controllable by the caller

### PR DESCRIPTION
**Pull request by AbdulBaseerKhan**
## Motivation
NAJ-64

## Description
Simple customisable logger added and all the internal logging of api should be done by it and have exposed the configureLogging so the end user can control logging accordingly.

## Checklist
- [x] Read the [contributing](https://github.com/near/near-api-js/blob/master/CONTRIBUTING.md) guidelines
- [x] Commit messages follow the [conventional commits](https://www.conventionalcommits.org/) spec
- [x] Performed a self-review of the PR
- [x] Added automated tests
- [x] Manually tested the change

### CODE REVIEW

File: packages/near-api-js/test/account.test.js [hunk 1]
```diff
@@ -75,19 +78,21 @@ test('findAccessKey returns the same access key when fetched simultaneously', as
...
-        console.log = function () {
-            logs.push(Array.from(arguments).join(' '));
-        };
+        configureLogging({
+            showLogs: () => true,
+            onLog: (args) => {
+                logs.push(Array.from(args).join(' '));
+            },
+            onWarn: () => {}, // Do nothing onWarn
+        });
```
Comment (esaminu) [inline packages/near-api-js/test/account.test.js:91]: I think you need to use spread in `args` here to recreate the same behavior as before

File: packages/near-api-js/test/account.test.js [hunk 2]
```diff
@@ -113,15 +117,18 @@ describe('with deploy contract', () => {
...
```
Comment (esaminu) [inline packages/near-api-js/test/account.test.js:127]: Same here, need to spread `args` in the `onLog` property to recreate old behavior

File: packages/near-api-js/test/promise.test.js [hunk 3]
```diff
@@ -28,15 +33,17 @@ describe('with promises', () => {
...
         logs = [];
-        console.log = function() {
-            logs.push(Array.from(arguments).join(' '));
-        };
+        configureLogging({
+            showLogs: () => true,
+            onLog: (args) => {
+                logs.push(Array.from(args).join(' '));
+            }, onWarn: () => {}
+        });
```
Comment (esaminu) [inline packages/near-api-js/test/promise.test.js:42]: Same as above with `args` spread in the `onLog` property

**Review by esaminu — COMMENTED** [review]
Looking good! I left some comments and a rebase is required, then we can land this 💯

**Comment by esaminu** [comment]
@AbdulBaseerKhan is this a breaking change btw? I see the PR title but I'm not sure how it's a breaking change 🙂

**Comment by marcinbodnar** [comment]
@AbdulBaseerKhan Could you please respond to Osman's comments? Thank you.

File: packages/near-api-js/test/account.test.js (hunk 1 above)
Comment (AbdulBaseerKhan) [reply packages/near-api-js/test/account.test.js:91]: @esaminu I have tested this behaviour and was working as the previous one.

Comment (esaminu) [reply packages/near-api-js/test/account.test.js:91]: It looks like it wouldn't work if multiple arguments are used. Does it still work in this case?

**Comment by denbite** [comment]
the idea of having an extendable logger was implemented in https://github.com/near/near-api-js/pull/1215

I'd suggest closing this one so we don't leave it out of date

**Pull request closed without merging**